### PR TITLE
[BEAM-14172] Update tox.ini for pydocs

### PR DESCRIPTION
**Pull request by aaltay**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/tox.ini [hunk 1]
```diff
@@ -143,6 +143,7 @@ deps =
   Sphinx==1.8.5
   sphinx_rtd_theme==0.4.3
   docutils<0.18
+  Jinja2==3.0.3 # TODO(BEAM-14712): Sphinx version is too old.
```
Comment (TheNeuralBit) [inline sdks/python/tox.ini:146]:
```suggestion
  Jinja2==3.0.3 # TODO(BEAM-14172): Sphinx version is too old.
```

**Review by TheNeuralBit — APPROVED** [review]
I think I tried updating sphinx at some point because I thought it would be easy and it turned out it was non-trivial for some reason (not super helpful I know..)

File: sdks/python/tox.ini (hunk 1 above)
Comment (aaltay) [reply sdks/python/tox.ini]: Thank you for catching this!

**Comment by aaltay** [comment]
> I think I tried updating sphinx at some point because I thought it would be easy and it turned out it was non-trivial for some reason (not super helpful I know..)

I looked at the version gap and did not attempt to upgrade. I think we need to re-purpose the bug to update sphinx.

**Pull request merged**